### PR TITLE
Fix ProtocolConformanceAnalysis for extensions

### PR DESCRIPTION
**Pull request by rajbarik**
https://github.com/apple/swift/pull/21025

### CODE REVIEW

**Comment by rajbarik** [comment]
@slavapestov @gottesmm Here is the PR.

**Comment by gottesmm** [comment]
LGTM. Thanks for including the test case!

**Comment by rajbarik** [comment]
Can we push this forward? @slavapestov @gottesmm

File: test/Interpreter/existential_transform.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift -O -wmo %s -o %t/a.out
+// RUN: %target-codesign %t/a.out
+// RUN: %target-run %t/a.out | %FileCheck %s
+// REQUIRES: executable_test
```
Comment (slavapestov) [inline test/Interpreter/existential_transform.swift]: This REQUIRES: objc_interop too. Maybe write this test without using Foundation though? Just define your own URL struct

File: test/SILOptimizer/existential_transform.swift [hunk 2]
```diff
@@ -1,5 +1,7 @@
 // RUN: %target-swift-frontend -O -sil-existential-specializer -Xllvm -sil-disable-pass=GenericSpecializer -Xllvm -sil-disable-pass=FunctionSignatureOpts -Xllvm -sil-disable-pass=SILCombine -emit-sil -sil-verify-all %s | %FileCheck %s
 
+import Foundation
```
Comment (slavapestov) [inline test/SILOptimizer/existential_transform.swift]: Can you write this test without using Foundation so that it runs on Linux too?

**Comment by rajbarik** [comment]
done @slavapestov  @gottesmm

File: test/Interpreter/existential_transform.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// REQUIRES: objc_interop
```
Comment (slavapestov) [inline test/Interpreter/existential_transform.swift]: The test no longer requires objc_interop since it doesn't use Foundation

**Review by slavapestov — APPROVED** [review]

**Comment by rajbarik** [comment]
Ready?

**Comment by rajbarik** [comment]
@slavapestov @gottesmm anything else?

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by rajbarik** [comment]
@gottesmm @slavapestov please run tests here. My mistake, the earlier one had a problem in the test checking.

**Comment by atrick** [comment]
@swift-ci test.

**Comment by atrick** [comment]
@swift-ci test source compatibility.

**Comment by rajbarik** [comment]
@atrick can you run smoke test here? Thanks.

**Comment by atrick** [comment]
@swift-ci smoke test.

**Pull request merged**